src/cpp/subscriber/StatisticsParticipantListener.cpp

```diff
@@ -318,12 +322,34 @@ void StatisticsParticipantListener::on_data_reader_discovery(
         }
     }
 
+    // Create IDL representation of the discovered type
+        // Get remote type information
+        xtypes::TypeObject remote_type_object;
+        if (RETCODE_OK != DomainParticipantFactory::get_instance()->type_object_registry().get_type_object(
+                    info.type_information.type_information.complete().typeid_with_size().type_id(),
+                    remote_type_object))
+        {
+            // Error
+            return;
+        }
+
+        // Build remotely discovered type
+        DynamicType::_ref_type remote_type = DynamicTypeBuilderFactory::get_instance()->create_type_w_type_object(
+            remote_type_object)->build();
+
+        // Serialize DynamicType into its IDL representation
+        std::stringstream idl;
+        idl_serialize(remote_type, idl);
+        discovery_info.type_idl = idl.str();
+
     entity_queue_->push(timestamp, discovery_info);
 
     // Wait until the entity queue is processed and restart the data queues
     entity_queue_->flush();
     data_queue_->start_consumer();
     monitor_service_status_data_queue_->start_consumer();
+
+    
```

Comment:
Remove extra line break
```suggestion
    monitor_service_status_data_queue_->start_consumer();
```

Reply:
Pending